app/src/main/java/kr/yjkim/book_search/data/BookSearchRepository.kt

```diff
@@ -1,27 +1,28 @@
 package kr.yjkim.book_search.data
 
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
 import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
 import kr.yjkim.book_search.data.network.KakaoService
 import kr.yjkim.book_search.data.network.RetrofitClient
-import kr.yjkim.book_search.data.schema.BookItem
+import kr.yjkim.book_search.util.ResultUiState
 
 object BookSearchRepository {
 
     private val kakaoService: KakaoService = RetrofitClient.kakaoService
 
-    private val _searchResult: MutableLiveData<Result<List<BookItem>>> = MutableLiveData()
-    val searchResult: LiveData<Result<List<BookItem>>> get() = _searchResult
+    private val _searchResult: MutableStateFlow<ResultUiState> = MutableStateFlow(ResultUiState.Loading)
+    val searchResult: StateFlow<ResultUiState> get() = _searchResult
```

Comment:
중요한건 아닐 수 있지만, 이왕이면 StateFlow 로 변환해서 반환하는게 좋습니다.
변환은 `asStateFlow()` 함수를 사용하면 됩니다!

변환하는게 좋은 이유는 외부에서 강제로 MutableStateFlow 로 캐스팅해서 사용이 가능하기 때문입니다.
`asStateFlow()` 를 사용하면 내부적으로 `ReadonlyStateFlow` 로 감싸기 때문에 안전해집니다.

```suggestion
    val searchResult: StateFlow<ResultUiState> get() = _searchResult.asStateFlow()
```